ios/engine/KMEI/KeymanEngine/Classes/Errors/SentryManager.swift

```diff
@@ -55,13 +57,29 @@ public class SentryManager {
     }
   }
 
+  public static var enabled: Bool {
+    get {
+      return !SentryManager.silenced
+    }
+
+    set(flag) {
+      SentryManager.silenced = !flag
+    }
+  }
+
   // Note:  this function is called from a separate thread!
+  // For some reason, does not update when `enabled` is changed unless the app is restarted.
```

Comment:
I mean, it's _literally_ exactly this.

If the app is loaded while the toggle set to enable error reporting, any Swift-side errors will always be reported while the app is loaded.  Even if the toggle's value is changed to disable reporting.  (Keyboard errors will be disabled immediately, though.)

Of course, if the app is closed and then reloaded after being toggled to disable errors, this also holds; as the toggle is disabled on the new app load, no Swift-side errors will be reported for the app's lifetime in this case, even if the toggle is then enabled.  (Keyboard errors, on the other hand, will be instantly enabled.)

It's almost like Sentry produces some sort of 'closure' during the app's init and refuses to accept changes to the enablement state within the function.  There's no clear reason _why_ it's ignoring the change in value... but it's ignoring that change all the same.

Reply:
I think we probably need to get to the bottom of this... 😭

Reply:
The check for initializing Sentry only happens on app load. So when the user toggles it, it wouldn't take effect till the app restarts, right?

Reply:
It's not that.

https://github.com/keymanapp/keyman/blob/f5fc278aa672d5f390f79df0d96f373477da71d8/ios/engine/KMEI/KeymanEngine/Classes/Errors/SentryManager.swift#L70-L79

The switch, among other things, changes the value of `SentryManager.silenced` there.  Except, for some odd reason, Sentry acts as if _it doesn't_.

Other relevant details:

https://github.com/keymanapp/keyman/blob/f5fc278aa672d5f390f79df0d96f373477da71d8/ios/engine/KMEI/KeymanEngine/Classes/Errors/SentryManager.swift#L17

https://github.com/keymanapp/keyman/blob/f5fc278aa672d5f390f79df0d96f373477da71d8/ios/engine/KMEI/KeymanEngine/Classes/Errors/SentryManager.swift#L53-L57

1.  At initialization time, we provide Sentry with a closure that may be used to determine whether or not to report errors.
2. That closure's code refers to a `static` 'flag' variable that may be dynamically altered at run-time - a variable outside of the closure.
3.  Yet, when the variable's value is changed at runtime, the closure acts as if it isn't changed.

Reply:
Have you tried adding some logging to watch this behaviour? Is there anything I can assist with? I think this is the only remaining PR before beta.

Reply:
I'm currently trying to figure out what sort of logging would help.

One thing I _have_ tried:  I tried replacing the closure with a newly-constructed one whenever the error-reporting setting was changed, just in case it somehow was being closure-captured.  No change in behavior.

Reply:
Can you log the value of _enabled and the address of SentryManager along with the thread doing the call? Just speculating here.

Reply:
Logged as #4126